src/BufferedRequest.ts

```diff
@@ -0,0 +1,38 @@
+import { Headers } from 'headers-polyfill/lib'
+import { decodeBuf } from './utils/bufferCodec'
+import { uuidv4 } from './utils/uuid'
+
+export class BufferedRequest {
```

Comment:
I'd call this `IsomorphicRequest` or just `Request`. We are really aiming at Fetch API's `Request` compatibility so whether we handle request body as a buffer is an implementation detail of this class. The consumer should not know about that.

Reply:
Calling it request makes it weird at place where we are using fetch api as well. I'll call it Isomorphic request.. 

Or may be `InterceptedRequest` ? wdyt?

Reply:
I vote for `IsomorphicRequest`. It describes what it stands for the best—an origin-agnostic representation of a request. It's a middle ground to describe any intercepted request regardless of its origin: `http.ClientRequest`, `XMLHttpRequest`, `fetch.Request`, and, in the future, things like Undici's request. Maybe there's a better engineering term for this kind of abstraction but I grabbed the term isomorphic as in "having the same shape (morph)".